### ISSUE TITLE
config:set database/username not working.

### ISSUE THREAD

**Issue by serweb-labs** [bolt/bolt#4934]
of course i mean to "nut",  the command line utility
example: `$ php app/nut config:set -f config.yml database/username root`
this return "New value for database/username: root was successful. File updated.".

Nothing is written.
It works well with other "keys: values".

Can anyone reproduce this behavior?

**Comment by SahAssar** [comment]
Which version is this on? Which platform/OS?

**Comment by GawainLynch** [comment]
Yes, I think @rossriley did some work on that, but from memory it doesn't handle nested values but I might be wrong.

What I did for the [tests is use some PERL magic](https://github.com/bolt/bolt/blob/release/2.2/tests/travis/test-db-config#L26-L31) if that helps.

**Comment by SahAssar** [comment]
@GawainLynch I tested it on 2.2.17 and it actually handled nested values perfectly for me, so I don't think it's that.

**Comment by GawainLynch** [comment]
@SahAssar Tested on master and it didn't work for me… let me try again, 'cause :koala: PEBCAK :stuck_out_tongue_winking_eye:

**Comment by GawainLynch** [comment]
Yeah, PEBCAK… I can't spell :grin:

**Comment by serweb-labs** [comment]
¡Thanks!, the blessed regular expressions. sorry I'm late to provide more details.
